--- /Users/ploskon/tmp/fjcontrib-1.042/Nsubjettiness/AxesDefinition.hh	2019-07-18 05:58:16.000000000 -0700
+++ /Users/ploskon/devel/heppy/cpptools/src/fjcontrib/fjcontrib-1.042/Nsubjettiness/AxesDefinition.hh	2021-02-28 23:45:07.000000000 -0800
@@ -93,16 +93,17 @@
    /// not be used for iterative refining (since that is the job of MeasureDefinition).
    virtual std::vector<fastjet::PseudoJet> get_starting_axes(int n_jets,
                                                              const std::vector<fastjet::PseudoJet>& inputs,
-                                                             const MeasureDefinition * measure) const = 0;
+                                                             const MeasureDefinition * measure) const
+   { std::vector<fastjet::PseudoJet> v; return v; } // MP
    
    /// Short description of AxesDefinitions (and any parameters)
-   virtual std::string short_description() const = 0;
+   virtual std::string short_description() const {std::string s = "not implemented"; return s;} // MP
 
    /// Long description of AxesDefinitions (and any parameters)
-   virtual std::string description() const = 0;
+   virtual std::string description() const {std::string s = "not implemented"; return s;} // MP
    
    /// This has to be defined in all derived classes, and allows these to be copied around.
-   virtual AxesDefinition* create() const = 0;
+   virtual AxesDefinition* create() const {return (AxesDefinition*) 0x0;}; // MP
    
 public:
    
@@ -186,7 +187,8 @@
    /// Destructor
    virtual ~AxesDefinition() {};
    
-protected:
+// protected:
+   public: // MP
    
    /// Default constructor contains no information.  Number of passes has to be set
    /// manually by derived classes using setNPass function.
