*** /Users/ploskon/devel/fjpy/recursivetools/include/RecursiveTools/RecursiveSymmetryCutBase.hh	2019-06-18 16:49:28.000000000 -0700
--- /Users/ploskon/devel/fjcmake/recursivetools/include/RecursiveTools/RecursiveSymmetryCutBase.hh	2019-06-14 05:01:04.000000000 -0700
***************
*** 220,225 ****
--- 220,226 ----
  protected:
    // the methods below have to be defined by deerived classes
    //----------------------------------------------------------------------
+ #ifndef SWIGPYTHON
    /// the cut on the symmetry measure (typically z) that one need to
    /// apply for a given pair of subjets p1 and p2
    virtual double symmetry_cut_fn(const PseudoJet & /* p1 */, 
***************
*** 227,233 ****
                                   void *extra_parameters = 0) const = 0;
    /// the associated dwescription
    virtual std::string symmetry_cut_description() const = 0;
! 
    //----------------------------------------------------------------------
    /// this defines status codes when checking for substructure
    enum RecursionStatus{
--- 228,239 ----
                                   void *extra_parameters = 0) const = 0;
    /// the associated dwescription
    virtual std::string symmetry_cut_description() const = 0;
! #else
!   virtual double symmetry_cut_fn(const PseudoJet & /* p1 */, 
!                                  const PseudoJet & /* p2 */,
!                                  void *extra_parameters = 0) const {return 0.0;};
!   virtual std::string symmetry_cut_description() const {std::string s = "not implemented"; return s;}
! #endif
    //----------------------------------------------------------------------
    /// this defines status codes when checking for substructure
    enum RecursionStatus{
